### PR TITLE
Reduce the time cost of fetching tileable data

### LINKED ISSUE #2592: [BUG] Decref raised connection refused  error
**Describe the bug**
Client raised connection refused when decref tileables, it happened when the tileable was executed in multiple sessions and one of them was invalid.

image

### PR DESCRIPTION
**Pull request by hekaisheng**
## What do these changes do?

Implement batch handlers for web APIs for meta and storage services to reduce the time of fetching data.

## Related issue number

Fixes #2592 

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/utils.py [hunk 1]
```diff
@@ -1085,27 +1085,38 @@ def arrow_array_to_objects(
     return obj
 
 
+_set_current_session = False
```
Comment (wjsi) [inline mars/utils.py:1088]: Rename to `_is_current_session_set`

File: mars/utils.py [hunk 2]
```diff
@@ -1085,27 +1085,38 @@ def arrow_array_to_objects(
...
-        session = ctx.get_current_session()
-        prev_default_session = get_default_session()
-        session.as_default()
+        reset_session = False
+        with AbstractSession._lock:
+            if not _set_current_session:
+                session = ctx.get_current_session()
+                prev_default_session = get_default_session()
+                session.as_default()
+                _set_current_session = True
```
Comment (wjsi) [inline mars/utils.py:1107]: Is it possible to handle nested `enter_default_session` calls?

**Review by wjsi — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**